samples/unit-tests/stock-tools/gui/demo.js

```diff
@@ -61,3 +61,115 @@ QUnit.test('Stocktools GUI', function (assert) {
     );
     */
 });
+
+QUnit.test('Disabling and Enabling stock Tools buttons, when series are invisible, #14192', function (assert) {
+    var wasInitCalled = false;
+
+    // Creating test controller and adding styles to button.
+    var toolsContainer = document.createElement('div');
+    toolsContainer.className += 'tools-container';
+    var button = document.createElement('button');
+    button.className += 'dummy-button';
+    button.innerHTML = 'dummy button';
+
+    toolsContainer.appendChild(button);
+
+    document.getElementById('container').parentNode.insertBefore(toolsContainer, document.getElementById('container'));
+
+    var chart = Highcharts.stockChart('container', {
+        stockTools: {
+            gui: {
+                enabled: false // disable the built-in toolbar
+            }
+        },
+        navigation: {
+            bindings: {
+                dummyButton: {
+                    className: 'dummy-button',
+                    init: function () {
+                        wasInitCalled = true;
+                    }
+                }
+            },
+            bindingsClassName: 'tools-container'
+        }
+    });
+
+    var controller = new TestController(chart);
+    toolsContainer.style.position = 'absolute';
+    toolsContainer.style['z-index'] = 99999;
+
+    controller.click(10, 10);
+    assert.equal(wasInitCalled, false, 'Init function should not be executed, when there is no series.');
+
+    wasInitCalled = false;
+    chart.addSeries({
+        id: 'aapl',
+        data: [
+            [0, 1, 2, 3, 4],
+            [1, 1, 2, 3, 4],
+            [2, 1, 2, 3, 4],
+            [3, 1, 2, 3, 4],
+            [4, 1, 2, 3, 4],
+            [5, 1, 2, 3, 4],
+            [5, 1, 2, 3, 4]
+        ]
+    });
+
+    controller.click(10, 10);
+    assert.equal(wasInitCalled, true, 'Init function should be executed, after series was added.');
+
+    chart.series[0].setVisible(false);
+    wasInitCalled = false;
+    controller.click(10, 10);
+    assert.equal(wasInitCalled, false, 'Init function should not be called, when series are invisible.');
+
+    chart.series[0].setVisible(true);
+    wasInitCalled = false;
+    controller.click(10, 10);
+    assert.equal(wasInitCalled, true, 'Init function should not be called, when series are visible.');
+
+    chart.series[0].remove();
+    wasInitCalled = false;
+    controller.click(10, 10);
+    assert.equal(wasInitCalled, false, 'Init function should not be called, after deleting the series.');
+
+    chart.addSeries({
+        id: 'aapl',
+        data: [
+            [0, 1, 2, 3, 4],
+            [1, 1, 2, 3, 4],
+            [2, 1, 2, 3, 4],
+            [3, 1, 2, 3, 4],
+            [4, 1, 2, 3, 4],
+            [5, 1, 2, 3, 4],
+            [5, 1, 2, 3, 4]
+        ]
+    });
```

Comment:
You are adding a default `line` series with `ohlc` data. I would stick with `line` series and simplify data.